Lib/test/test_dataclasses.py

```diff
@@ -1300,6 +1300,21 @@ class D:
         self.assertTrue(is_dataclass(d.d))
         self.assertFalse(is_dataclass(d.e))
 
+    def test_is_dataclass_when_getattr_always_returns(self):
+        # See bpo-37868.
+        class A:
+            def __getattr__(self, key):
+                return 0
+
+        self.assertFalse(is_dataclass(A()))
+        self.assertFalse(is_dataclass(A))
```

Comment:
Maybe add also indirect tests for `_is_dataclass_instance()`?

`asdict()`, `astuple()`, `replace()`

Reply:
Done.